src/plugins/query_enhancements/server/search/sql_async_search_strategy.ts

```diff
@@ -39,78 +43,39 @@ export const sqlAsyncSearchStrategyProvider = (
   return {
     search: async (context, request: any, options) => {
       try {
-        const query: Query = request?.body?.query;
-        // Create job: this should return a queryId and sessionId
-        if (query) {
-          const df = request.body?.df;
-          request.body = {
-            query: query.query,
-            datasource: query.dataset?.dataSource?.title,
-            lang: SEARCH_STRATEGY.SQL,
-            sessionId: df?.meta?.sessionId,
-          };
-          const rawResponse: any = await sqlAsyncFacet.describeQuery(context, request);
-          // handles failure
-          if (!rawResponse.success) {
-            return {
-              type: DATA_FRAME_TYPES.POLLING,
-              body: { error: rawResponse.data },
-              took: rawResponse.took,
-            } as IDataFrameError;
-          }
-          const queryId = rawResponse.data?.queryId;
-          const sessionId = rawResponse.data?.sessionId;
+        const query: Query = request.body.query;
+        const startTime = Date.now();
+        request.body = { ...request.body, lang: SEARCH_STRATEGY.SQL };
+        const rawResponse: any = await sqlAsyncFacet.describeQuery(context, request);
+
+        if (!rawResponse.success) handleFacetError(rawResponse);
+
+        const statusConfig = buildQueryStatusConfig(rawResponse);
+        request.params = { queryId: statusConfig.queryId };
+
+        const response = await handleQueryStatusPolling(async () => {
+          logger.info(`sqlAsyncSearchStrategy: query job: ${statusConfig.queryId}`);
+          return sqlAsyncJobsFacet.describeQuery(context, request);
```

Comment:
Not a blocker for this PR: Need to handle handle two types of error states here:
1. From sql -> 503 when sending request query
2. From Flint/Spark EMR job -> 200 with status as "FAILED" in response.data object